library/formatting.inc.php

```diff
@@ -216,3 +216,24 @@ function oeFormatAge($dobYMD, $nowYMD = '', $format = 0)
 
     return $age;
 }
+
+
+/**
+ * Returns the complete formatted stae string according the global date format
+ * @param $timestamp
+ * @return bool|string
+ */
+function oeFormatLongDate($timestamp)
+{
+    if(!$timestamp){
+        $timestamp = date('Y-m-d H:i');
+    }
+    // assume input is yyyy-mm-dd
+    if ($GLOBALS['date_display_format'] == 1) {      // mm/dd/yyyy, note year is added below
+        $newDate = date('m-d-Y H:i',$timestamp);
```

Comment:
Does this work?
Isn't the second parameter supposed to be a time stamp?
Agree this function is needed, but would be better off feeding the first 10 character to oeFormatShortDate() and then the remaining characters to oeFormatTime(). Then will be using the same algorithms that are already used.